testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/transaction/TransactionStatisticsTestCase.java

```diff
@@ -0,0 +1,167 @@
+    private static final String GROUP_STATISTICS_ATTR = "statistics";
+
+    static class TransactionEnabledSetup implements ServerSetupTask {
+        private ModelNode statisticsEnabledOriginValue;
+        @Override
+        public void setup(ManagementClient managementClient, String s) throws Exception {
+            ModelNode opResult = MaximumTimeoutTestCase.executeForResult(
+                    managementClient.getControllerClient(), Util.getReadAttributeOperation(TX_SUBSYSTEM_ADDRESS, STATISTICS_ENABLED_ATTR));
+            // value can be either expression or boolean, let's tackle it with try/catch
+            try {
+                ValueExpression originalAsExpression = opResult.asExpression();
+                statisticsEnabledOriginValue = new ModelNode().set(originalAsExpression);
+            } catch (Exception notAnExpression) {
+                // let's try to use boolean
+                boolean originalAsBoolean = opResult.asBoolean();
+                statisticsEnabledOriginValue = new ModelNode().set(originalAsBoolean);
+            }
+            writeStatisticsEnabled(managementClient, new ModelNode().set(true));
+        }
+        @Override
+        public void tearDown(ManagementClient managementClient, String s) throws Exception {
+            writeStatisticsEnabled(managementClient, statisticsEnabledOriginValue);
+        }
+        private void writeStatisticsEnabled(ManagementClient client, ModelNode modelNodeWriteData) {
+            ModelNode op = Util.getWriteAttributeOperation(TX_SUBSYSTEM_ADDRESS, STATISTICS_ENABLED_ATTR, modelNodeWriteData);
+            MaximumTimeoutTestCase.executeForResult(client.getControllerClient(), op);
+        }
+    }
+
+
+    @Resource(mappedName = "java:/TransactionManager")
+    private static TransactionManager tm;
+
+    @ArquillianResource
+    private ManagementClient managementClient;
+
+    @Deployment
+    public static JavaArchive createDeployment() {
+        return ShrinkWrap.create(JavaArchive.class)
+                .addClasses(TransactionStatisticsTestCase.class, TransactionStatisticsTestCase.TransactionEnabledSetup.class,
+                        MaximumTimeoutTestCase.class)
+                .addPackage(TestXAResource.class.getPackage())
+                .addAsManifestResource(new StringAsset("Dependencies: org.jboss.as.controller, org.jboss.remoting\n"), "MANIFEST.MF")
+                .addAsManifestResource(createPermissionsXmlAsset(
+                    // ManagementClient needs the following permissions and a dependency on 'org.jboss.remoting3' module
+                    new RemotingPermission("createEndpoint"),
+                    new RemotingPermission("connect"),
+                    new FilePermission(System.getProperty("jboss.inst") + "/standalone/tmp/auth/*", "read")
+                ), "permissions.xml");
+    }
+
+    @Test
+    public void transactionStatisticsGrouped() throws Exception{
```

Comment:
Why did you only test the following attributes

```
number-of-transactions
number-of-committed-transactions
number-of-aborted-transactions
number-of-application-rollbacks
```

but not these ones:

```
average-commit-time
number-of-heuristics
number-of-inflight-transactions
number-of-nested-transactions
number-of-resource-rollbacks
number-of-system-rollbacks
number-of-timed-out-transactions
```

Reply:
Agree, it would be more appropriate to test all the properties being in group. And possibly trying to check if other statistics work.
I adjusted the PR to work with more statistics (those which are available without difficulties) and check if all of them are listed in the group. Can you  @mmusgrov look at again, please?